core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -219,6 +236,13 @@ private[spark] class Executor(
     heartbeater.shutdown()
     heartbeater.awaitTermination(10, TimeUnit.SECONDS)
     threadPool.shutdown()
+
+    // Notify plugins that executor is shutting down so they can terminate cleanly
+    // oldContextClassLoader is redefined in case the class loader changed during execution
+    val oldContextClassLoader = Utils.getContextOrSparkClassLoader
```

Comment:
I think it's worth it to add a method in `Utils.scala` to do the class loader thingy. This code pattern is used in a bunch of places, it would be nice to make it easier (and later change existing code to use the new method).

Reply:
Do you mean make a function that does something like 
`Utils.withClassLoader(replClassLoader) {
load plugins
}` ?

Reply:
Yes.